token-lending/cli/src/main.rs

```diff
@@ -0,0 +1,700 @@
+        .get_minimum_balance_for_rent_exemption(Token::LEN)?;
+    let liquidity_supply_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Token::LEN)?;
+    let liquidity_fee_receiver_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Token::LEN)?;
+    let liquidity_host_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Token::LEN)?;
+
+    let total_balance = reserve_balance
+        + collateral_mint_balance
+        + collateral_supply_balance
+        + user_collateral_balance
+        + liquidity_supply_balance
+        + liquidity_fee_receiver_balance
+        + liquidity_host_balance;
+
+    let mut create_accounts_transaction_1 = Transaction::new_with_payer(
+        &[
+            create_account(
+                &config.fee_payer.pubkey(),
+                &reserve_keypair.pubkey(),
+                reserve_balance,
+                Reserve::LEN as u64,
+                &spl_token_lending::id(),
+            ),
+            create_account(
+                &config.fee_payer.pubkey(),
+                &collateral_mint_keypair.pubkey(),
+                collateral_mint_balance,
+                Mint::LEN as u64,
+                &spl_token::id(),
+            ),
+            create_account(
+                &config.fee_payer.pubkey(),
+                &collateral_supply_keypair.pubkey(),
+                collateral_supply_balance,
+                Token::LEN as u64,
+                &spl_token::id(),
+            ),
+            create_account(
+                &config.fee_payer.pubkey(),
+                &user_collateral_keypair.pubkey(),
+                user_collateral_balance,
+                Token::LEN as u64,
+                &spl_token::id(),
+            ),
+        ],
+        Some(&config.fee_payer.pubkey()),
+    );
+
+    let mut create_accounts_transaction_2 = Transaction::new_with_payer(
+        &[
+            create_account(
+                &config.fee_payer.pubkey(),
+                &liquidity_supply_keypair.pubkey(),
+                liquidity_supply_balance,
+                Token::LEN as u64,
+                &spl_token::id(),
+            ),
+            create_account(
+                &config.fee_payer.pubkey(),
+                &liquidity_fee_receiver_keypair.pubkey(),
+                liquidity_fee_receiver_balance,
+                Token::LEN as u64,
+                &spl_token::id(),
+            ),
+            create_account(
+                &config.fee_payer.pubkey(),
+                &liquidity_host_keypair.pubkey(),
+                liquidity_host_balance,
+                Token::LEN as u64,
+                &spl_token::id(),
+            ),
+        ],
+        Some(&config.fee_payer.pubkey()),
+    );
+
+    let mut init_reserve_transaction = Transaction::new_with_payer(
+        &[
+            approve(
+                &spl_token::id(),
+                &source_liquidity_pubkey,
```

Comment:
This requires the owner of `source_liquidity` to sign but I don't see that listed as a signer

Reply:
I think this works given the provided instructions because the owner of `source_liquidity` happens to be the same as the fee payer that makes the transfer, and sometimes also the lending market owner. Should the command itself require another signer then?

Reply:
Yeah there ends up being lots of overlap. You may want to look at the spl-token-cli to see how it handles signers (it's complex)

Reply:
Done. I did this somewhat simply by just providing additional keypairs rather than worrying about signers that could be external. I'm thinking the latter could be added later.